### PR TITLE
[4.0] Increase the Swift ABI version to 5

### PR DESCRIPTION
**Pull request by aschwaighofer**
Explanation: We need to increase the Swift ABI version number such that tools can recognize that object files generated by older compilers are not compatible.

Scope: Object files generated by older compiler versions will not be compatible. This is intended since they are not ABI compatible.

Radar (and possibly SR Issue): rdar://32164125

Risk: Low. Incompatible objects might be discovered and need to be recompiled. This is good since they are likely incompatible.

Testing: Tested by regression test suite.

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Comment by aschwaighofer** [comment]
@gparker42 Can you review this change?

File: test/IRGen/objc.swift [hunk 1]
```diff
@@ -143,5 +143,5 @@ class WeakObjC {
 // CHECK:  i32 1, !"Objective-C Image Info Version", i32 0}
 // CHECK:  i32 1, !"Objective-C Image Info Section", !"__DATA, __objc_imageinfo, regular, no_dead_strip"}
 //   512 == (2 << 8).  2 is the Swift ABI version.
```
Comment (gparker42) [inline test/IRGen/objc.swift]: You should update this comment (it has been out-of-date for some time).

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Comment by gparker42** [comment]
Okay, looks good.

**Pull request merged**